sycl/include/sycl/detail/common.hpp

```diff
@@ -35,7 +35,9 @@ using EnableIfOutputPointerT = std::enable_if_t<
 
 template <typename DataT>
 using EnableIfOutputIteratorT = std::enable_if_t<
-    /*is_output_iterator<DataT>::value &&*/ !std::is_pointer_v<DataT>>;
+    /*is_output_iterator<DataT>::value &&*/ !std::is_pointer_v<DataT> &&
+    !std::is_const_v<std::remove_reference_t<
```

Comment:
1) Line 34 has the same bug.
2) Have you checked if comment at line 30 is up-to-date?

Reply:
I will check if its prossible to uncomment the is_output_iterator trait then I'll get back to you. If this is unblocked then this PR is done.

Reply:
@aelovikov-intel It looks like the problem was with void* ptr so I added a full specialization for that case. This enabled me to uncomment the is_output_iterator